lib/workload/stateless/stacks/icav2-event-translator/translator_service/tests/test_icav2_event_translator.py

```diff
@@ -0,0 +1,223 @@
+import json
+import os
+import datetime 
+import unittest
+import botocore.session
+from botocore.stub import Stubber
+from unittest.mock import patch
+from freezegun import freeze_time
+from uuid import UUID
+'''
+{
+    "portalRunId": "202405012397actg",
+    "timestamp": "2024-05-01T09:25:44Z",
+    "status": "succeeded",
+    "workflowType": "bssh_bcl_convert",
+    "workflowVersion": "4.2.7",
+    "payload": {
+      "refId": null,
+      "version": "0.1.0",
+      "projectId": "bxxxxxxxx-dxxx-4xxxx-adcc-xxxxxxxxx",
+      "analysisId": "aaaaafe8-238c-4200-b632-d5dd8c8db94a",
+      "userReference": "540424_A01001_0193_BBBBMMDRX5_c754de_bd822f",
+      "timeCreated": "2024-05-01T10:11:35Z",
+      "timeModified": "2024-05-01T11:24:29Z",
+      "pipelineId": "bfffffff-cb27-4dfa-846e-acd6eb081aca",
+      "pipelineCode": "BclConvert v4_2_7",
+      "pipelineDescription": "This is an autolaunch BclConvert pipeline for use by the metaworkflow",
+      "pipelineUrn": "urn:ilmn:ica:pipeline:bfffffff-cb27-4dfa-846e-acd6eb081aca#BclConvert_v4_2_7"
+    },
+    "serviceVersion": "0.1.0"
+  }
+'''
+
+# Assuming you adapt your lambda function to use botocore for creating clients
+from icav2_event_translator import handler  # Import your lambda function module
+
+class TestICAv2EventTranslator(unittest.TestCase):
```

Comment:
Nice! Great job for unit testing.

Reply:
Yes, nice! This might be a good one rather than relying on external library (`localstack`) 😅

Reply:
_you don't like localstack 💔 ..._ 😿

Reply:
I don't like localstack either! ;P